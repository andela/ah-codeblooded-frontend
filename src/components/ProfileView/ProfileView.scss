.profile-view {
  padding: .5em;
  min-width: 30px;

  .card-title {
    opacity: 1;
  }

  .card-content {
    max-height: 100px;
    min-height: 100px;
  }

  .followButton {
    position: sticky;
  }

  .card-image {
    min-height: 300px;
  }
}
