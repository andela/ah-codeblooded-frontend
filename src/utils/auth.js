export const getCurrentUser = () => {
  // read the user details from local storage
  const user = localStorage.getItem('user');
  return user ? JSON.parse(user) : null;
};
export const getToken = () => (getCurrentUser() ? getCurrentUser().token : null);
