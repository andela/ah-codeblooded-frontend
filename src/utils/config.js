const config = {
  BASE_URL: process.env.API_URL,
};

export default config;
