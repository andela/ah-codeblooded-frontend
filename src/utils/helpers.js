export const randomKey = () => (
  Math.floor((Math.random() * 100) + 1)
);

export const getQueryParam = (param) => {
  const urlParams = new URL(window.location.href).searchParams;
  return urlParams.get(param);
};


export const getLocation = () => window.location.toString();

export const reloadLocation = () => window.location.assign(window.location.href);
