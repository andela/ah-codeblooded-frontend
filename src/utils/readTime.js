export default (seconds) => {
  if (seconds < 60) {
    const s = Math.round(seconds);
    return `${s + (s === 1 ? ' Second' : ' Seconds')} read`;
  } if (seconds < 3600) {
    const minutes = Math.round(seconds / 60);
    return `${minutes + (minutes === 1 ? ' Minute' : ' Minutes')} read`;
  }
  return 'More than 1 hour';
};
