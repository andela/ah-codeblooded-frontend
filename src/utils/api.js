import axios from 'axios';
import config from './config';
import { getToken } from './auth';
import store from '../store';
import { NETWORK_ERROR_DETECTED, NO_NETWORK_ERROR } from "../containers/NetworkPopup/state/types";

const token = getToken();

const uri = config.BASE_URL;

export const getURL = endpoint => `${uri}${endpoint}`;

const api = axios.create({
  baseURL: uri,
});

api.interceptors.request.use(
  (cfg) => {
    if (token) {
      cfg.headers.authorization = `Token ${token}`;
    }
    return cfg;
  },
);

api.interceptors.response.use(
  (response) => {
    store.dispatch({ type: NO_NETWORK_ERROR });
    return response;
  },
  (error) => {
    if (!error.response) {
      store.dispatch({ type: NETWORK_ERROR_DETECTED });
    }
    return Promise.reject(error);
  },
);

api.interceptors.response.use(
  response => response,
  (error) => {
    if (error.response && error.response.status === 403) {
      localStorage.setItem('user', null);
      window.location.assign(`${window.location.protocol}//${window.location.host.toString()}/login`);
    }
    return Promise.reject(error);
  },
);
export default api;
