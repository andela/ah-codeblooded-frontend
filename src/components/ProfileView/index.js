import React from 'react';
import profileImage from '../../assets/images/profile.jpg';
import './ProfileView.scss';
import ConnnectedFollowUnfollow from "../../containers/FollowUnfollow";
import { getCurrentUser } from "../../utils/auth";

const ProfileView = ({ profile }) => (
  <>
    <div className="profile-view">
      <div className="row">
        <div className="card">
          <div className="card-image">
            <img src={profile.image || profileImage} alt="" />
            <span className="card-title"><a className="green-text" href={profile && `/profiles/view/${profile.username}`}>{ profile.username }</a></span>
          </div>
          <div className="card-content">
            <div className="followButton">
              <ConnnectedFollowUnfollow username={profile.username} user={getCurrentUser()} />
            </div>
            <br />
            <br />
          </div>
        </div>
      </div>
    </div>
    </>
);

ProfileView.defaultProps = {
  profile: {},
  following: false,
  handleFollow: null,
};

export default ProfileView;
