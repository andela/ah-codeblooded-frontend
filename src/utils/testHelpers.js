import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import api from './api';
import editorState from '../pages/Articles/Create/state/editorState';
import { initialState } from '../pages/Articles/state/reducer';

const middlewares = [thunk];
export const mockStore = configureMockStore(middlewares);
export const axiosMock = new MockAdapter(api);

export const testReducer = (action, value, result, reducer, state) => {
  const newState = reducer(state, action);
  expect(newState[value])
    .toEqual(result);
};

export const user = {
  username: 'gitaumoses4',
  image: 'https://avatars3.githubusercontent.com/u/25629064?s=460&v=4',
  email: 'gitaumoses4@gmail.com',
  password: 'Password1!',
};

export const article = {
  title: 'The Dungeon Dragon',
  body: JSON.stringify(editorState),
  description: 'The Dungeon Dragon Story',
  slug: 'the-dungeon-dragon-12jhgf323',
  tags: ['dragons', 'dungeon'],
  published: false,
  author: user,
};

export const comment = {
  id: 8,
  body: "Here we go",
  author: {
    username: "gitaumoses4",
    bio: "This is my story, this is my song. Praising my saviour all thy day long",
    image: "https://res.cloudinary.com/codeblooded/image/upload/v1542154278/lwwun6s61usi3jnxjulz.jpg",
  },
  likes: {
    count: 0,
    me: false,
  },
  dislikes: {
    count: 0,
    me: false,
  },
  parent: null,
  created_at: "2018-11-21T23:41:34.216533Z",
};

export const articleCRUD = {
  ...initialState,
};


export const checkSnapshot = (component) => {
  const snap = shallow(component);
  expect(snap).toMatchSnapshot();
};

export const loginUser = (usr = user) => {
  localStorage.setItem("user", JSON.stringify(usr));
};
