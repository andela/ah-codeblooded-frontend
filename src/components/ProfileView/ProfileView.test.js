import React from 'react';
import ProfileView from './index';
import { checkSnapshot, user } from '../../utils/testHelpers';

describe('<ProfileView/> component', () => {
  it('should render an <ProfileView/> component', () => {
    checkSnapshot(<ProfileView user={user} />);
  });
});
