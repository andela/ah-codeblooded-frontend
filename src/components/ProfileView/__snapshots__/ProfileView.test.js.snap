// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileView/> component should render an <ProfileView/> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileView
    following={false}
    handleFollow={null}
    profile={Object {}}
    user={
      Object {
        "email": "gitaumoses4@gmail.com",
        "image": "https://avatars3.githubusercontent.com/u/25629064?s=460&v=4",
        "password": "Password1!",
        "username": "gitaumoses4",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="profile-view"
      >
        <div
          className="row"
        >
          <div
            className="card"
          >
            <div
              className="card-image"
            >
              <img
                alt=""
                src="profile.jpg"
              />
              <span
                className="card-title"
              >
                <a
                  className="green-text"
                  href="/profiles/view/undefined"
                />
              </span>
            </div>
            <div
              className="card-content"
            >
              <div
                className="followButton"
              >
                <Connect(FollowUnfollow)
                  user={null}
                />
              </div>
              <br />
              <br />
            </div>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="card"
          >
            <div
              className="card-image"
            >
              <img
                alt=""
                src="profile.jpg"
              />
              <span
                className="card-title"
              >
                <a
                  className="green-text"
                  href="/profiles/view/undefined"
                />
              </span>
            </div>
            <div
              className="card-content"
            >
              <div
                className="followButton"
              >
                <Connect(FollowUnfollow)
                  user={null}
                />
              </div>
              <br />
              <br />
            </div>
          </div>
        </div>,
        "className": "profile-view",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card"
          >
            <div
              className="card-image"
            >
              <img
                alt=""
                src="profile.jpg"
              />
              <span
                className="card-title"
              >
                <a
                  className="green-text"
                  href="/profiles/view/undefined"
                />
              </span>
            </div>
            <div
              className="card-content"
            >
              <div
                className="followButton"
              >
                <Connect(FollowUnfollow)
                  user={null}
                />
              </div>
              <br />
              <br />
            </div>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-image"
              >
                <img
                  alt=""
                  src="profile.jpg"
                />
                <span
                  className="card-title"
                >
                  <a
                    className="green-text"
                    href="/profiles/view/undefined"
                  />
                </span>
              </div>,
              <div
                className="card-content"
              >
                <div
                  className="followButton"
                >
                  <Connect(FollowUnfollow)
                    user={null}
                  />
                </div>
                <br />
                <br />
              </div>,
            ],
            "className": "card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    src="profile.jpg"
                  />,
                  <span
                    className="card-title"
                  >
                    <a
                      className="green-text"
                      href="/profiles/view/undefined"
                    />
                  </span>,
                ],
                "className": "card-image",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "src": "profile.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <a
                      className="green-text"
                      href="/profiles/view/undefined"
                    />,
                    "className": "card-title",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                      "className": "green-text",
                      "href": "/profiles/view/undefined",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "a",
                  },
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="followButton"
                  >
                    <Connect(FollowUnfollow)
                      user={null}
                    />
                  </div>,
                  <br />,
                  <br />,
                ],
                "className": "card-content",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(FollowUnfollow)
                      user={null}
                    />,
                    "className": "followButton",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "user": null,
                      "username": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="profile-view"
        >
          <div
            className="row"
          >
            <div
              className="card"
            >
              <div
                className="card-image"
              >
                <img
                  alt=""
                  src="profile.jpg"
                />
                <span
                  className="card-title"
                >
                  <a
                    className="green-text"
                    href="/profiles/view/undefined"
                  />
                </span>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="followButton"
                >
                  <Connect(FollowUnfollow)
                    user={null}
                  />
                </div>
                <br />
                <br />
              </div>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="card"
            >
              <div
                className="card-image"
              >
                <img
                  alt=""
                  src="profile.jpg"
                />
                <span
                  className="card-title"
                >
                  <a
                    className="green-text"
                    href="/profiles/view/undefined"
                  />
                </span>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="followButton"
                >
                  <Connect(FollowUnfollow)
                    user={null}
                  />
                </div>
                <br />
                <br />
              </div>
            </div>
          </div>,
          "className": "profile-view",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card"
            >
              <div
                className="card-image"
              >
                <img
                  alt=""
                  src="profile.jpg"
                />
                <span
                  className="card-title"
                >
                  <a
                    className="green-text"
                    href="/profiles/view/undefined"
                  />
                </span>
              </div>
              <div
                className="card-content"
              >
                <div
                  className="followButton"
                >
                  <Connect(FollowUnfollow)
                    user={null}
                  />
                </div>
                <br />
                <br />
              </div>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-image"
                >
                  <img
                    alt=""
                    src="profile.jpg"
                  />
                  <span
                    className="card-title"
                  >
                    <a
                      className="green-text"
                      href="/profiles/view/undefined"
                    />
                  </span>
                </div>,
                <div
                  className="card-content"
                >
                  <div
                    className="followButton"
                  >
                    <Connect(FollowUnfollow)
                      user={null}
                    />
                  </div>
                  <br />
                  <br />
                </div>,
              ],
              "className": "card",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <img
                      alt=""
                      src="profile.jpg"
                    />,
                    <span
                      className="card-title"
                    >
                      <a
                        className="green-text"
                        href="/profiles/view/undefined"
                      />
                    </span>,
                  ],
                  "className": "card-image",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "src": "profile.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <a
                        className="green-text"
                        href="/profiles/view/undefined"
                      />,
                      "className": "card-title",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                        "className": "green-text",
                        "href": "/profiles/view/undefined",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                    "type": "span",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="followButton"
                    >
                      <Connect(FollowUnfollow)
                        user={null}
                      />
                    </div>,
                    <br />,
                    <br />,
                  ],
                  "className": "card-content",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(FollowUnfollow)
                        user={null}
                      />,
                      "className": "followButton",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "user": null,
                        "username": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
